### PR TITLE
Form changing by holding items

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
This PR implements form changes by holding and taking away items, both in the party menu and Pokémon Storage System.

Giving item from party menu.
gira form

Manipulating items in PSS
form arc

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Ready for review :)

**Comment by AsparagusEduardo** [comment]
Ok, now it's really ready xD

**Comment by AsparagusEduardo** [comment]
Put on hold until #1714 is merged

**Comment by BuffelSaft** [comment]
Found an edge case: if the player is given an Arceus holding a plate, it doesn't change form until the plate is removed and given back. Test case was using `givemon SPECIES_ARCEUS, 100, ITEM_SPLASH_PLATE`, Arceus was Arceus-normal when received. Not sure if this actually matters, but maybe species could be updated on opening the party menu to prevent edge cases like this one?

**Comment by AsparagusEduardo** [comment]
> Found an edge case: if the player is given an Arceus holding a plate, it doesn't change form until the plate is removed and given back. Test case was using `givemon SPECIES_ARCEUS, 100, ITEM_SPLASH_PLATE`, Arceus was Arceus-normal when received. Not sure if this actually matters, but maybe species could be updated on opening the party menu to prevent edge cases like this one?

I'm thinking of doing the change during `givemon`. When doing the initial version of this PR, I originally had it so it changed species when highlighted a mon on the pc, but that ended up causing a very slight bit of lag. Because of that I want to avoid spamming the function and only use it whenever is needed.

**Comment by AsparagusEduardo** [comment]
Ready for re-review.

File: src/pokemon.c
Comment (BuffelSaft) [inline src/pokemon.c:8057]:
```suggestion
// Returns SPECIES_NONE if no form change is possible
```

**Review by BuffelSaft — COMMENTED** [review]
pokeemerald-0
I've played around with a whole box of item form change mons and found no issues, they're working in battle, in the PC and the party. Style-wise the code is good, just need to capitalise that one comment and this is good to go I think.

**Comment by AsparagusEduardo** [comment]
Conflicts resolved. Ready for re-review.

**Comment by AsparagusEduardo** [comment]
Now they're actually resolved. Ready for re-review.

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**
